packages/contracts/test/token/erc20/governance-erc20.ts

```diff
@@ -0,0 +1,221 @@
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+
+import {
+  DAO,
+  GovernanceERC20,
+  GovernanceERC20__factory,
+} from '../../../typechain';
+import {deployNewDAO} from '../../test-utils/dao';
+import {OZ_ERRORS} from '../../test-utils/error';
+import {getInterfaceID} from '../../test-utils/interfaces';
+
+export type MintSettings = {
+  receivers: string[];
+  amounts: number[];
+};
+
+const MINT_PERMISSION_ID = ethers.utils.id('MINT_PERMISSION');
+const governanceERC20Name = 'GovernanceToken';
+const governanceERC20Symbol = 'GOV';
+
+describe('GovernanceERC20', function () {
+  let signers: SignerWithAddress[];
+  let dao: DAO;
+  let token: GovernanceERC20;
+  let GovernanceERC20: GovernanceERC20__factory;
+  let mintSettings: MintSettings;
+  let defaultInitData: [string, string, string, MintSettings];
+
+  before(async () => {
+    signers = await ethers.getSigners();
+    dao = await deployNewDAO(signers[0].address);
+    GovernanceERC20 = await ethers.getContractFactory('GovernanceERC20');
+  });
+
+  beforeEach(async function () {
+    mintSettings = {
+      receivers: signers.slice(0, 3).map(s => s.address),
+      amounts: [123, 456, 789],
+    };
+    defaultInitData = [
+      dao.address,
+      governanceERC20Name,
+      governanceERC20Symbol,
+      mintSettings,
+    ];
+  });
+
+  describe('initialize:', async () => {
+    it('reverts if trying to re-initialize', async () => {
```

Comment:
why deploying GovernanceERC20 in each test ? can't we put it in beforeEach of `describe` or the main beforeEach(the most top one)

Reply:
Done in 34b4588e60c6cd7ee5b274eadaa59752939a676e